hedera-mirror-rest/accounts.js

```diff
@@ -297,23 +358,78 @@ const getAccounts = async (req, res) => {
  * @return {Promise}
  */
 const getOneAccount = async (req, res) => {
-  // Validate query parameters first
-  utils.validateReq(req, acceptedSingleAccountParameters);
-
-  const encodedId = await EntityService.getEncodedId(req.params[constants.filterKeys.ID_OR_ALIAS_OR_EVM_ADDRESS]);
-
   // Parse the filter parameters for account-numbers, balance, and pagination
+  const filters = utils.buildAndValidateFilters(req.query, acceptedSingleAccountParameters);
+  const encodedId = await EntityService.getEncodedId(req.params[constants.filterKeys.ID_OR_ALIAS_OR_EVM_ADDRESS]);
   const parsedQueryParams = req.query;
-  const [tsQuery, tsParams] = utils.parseTimestampQueryParam(parsedQueryParams, 't.consensus_timestamp');
+  const timestampFilters = filters.filter((filter) => filter.key === filterKeys.TIMESTAMP);
+  const [tsRange, eqValues, neValues] = utils.checkTimestampRange(timestampFilters, false, true, true, false);
+  const [transactionTsQuery, transactionTsParams] = utils.buildTimestampQuery(
+    tsRange,
+    't.consensus_timestamp',
+    neValues,
+    eqValues
+  );
+
+  let paramCount = 0;
   const resultTypeQuery = utils.parseResultParams(req);
   const {query, params, order, limit} = utils.parseLimitAndOrderParams(req);
+  const accountIdParams = [encodedId];
+  const tokenBalanceQuery = {
+    query: `account_id = $${++paramCount}`,
+    params: accountIdParams,
+    limit: tokenBalanceResponseLimit.singleAccount,
+  };
+  const entityAccountQuery = {query: `e.id = $${paramCount}`, params: []};
+  const entityStakeQuery = {query: `(es.id = $${paramCount} OR es.id IS NULL)`, params: []};
+
+  const accountBalanceQuery = {query: '', params: []};
+  if (transactionTsQuery) {
+    let tokenBalanceTsQuery = transactionTsQuery
+      .replaceAll('t.consensus_timestamp', 'created_timestamp')
+      .replaceAll('?', (_) => `$${++paramCount}`);
+
+    tokenBalanceQuery.query += ` and ${tokenBalanceTsQuery} `;
+    tokenBalanceQuery.params = tokenBalanceQuery.params.concat(transactionTsParams);
+
+    const [entityTsQuery, entityTsParams] = utils.buildTimestampRangeQuery(
+      tsRange,
+      Entity.getFullName(Entity.TIMESTAMP_RANGE),
+      neValues,
+      eqValues
+    );
+    entityAccountQuery.query += ` and ${entityTsQuery.replaceAll('?', (_) => `$${++paramCount}`)}`;
+    entityAccountQuery.params = entityAccountQuery.params.concat(entityTsParams);
+
+    const [balanceFileTsQuery, balanceFileTsParams] = utils.buildTimestampQuery(
+      tsRange,
+      'consensus_timestamp',
+      [],
+      eqValues,
+      false
+    );
+    const accountBalanceTs = await balances.getAccountBalanceTimestamp(
+      balanceFileTsQuery.replaceAll(opsMap.eq, opsMap.lte),
+      balanceFileTsParams,
+      order
+    );
+    const timestampExcluded = neValues.reduce((prev, curr) => prev || curr === accountBalanceTs, false);
```

Comment:
I see the neValues is an array of `BigInt` now, however, `accountBalanceTs` can be either a `Number` or a `BigInt`, so we'll end up with `10n === 10` evaluated to false, we can do either of the two

- use `utils.parseInteger` instead of `BigInt` in `checkTimestampRange`
- use `curr == accountBalanceTs` here and let js do the coercion

in addition, we should do `Array.prototype.some` instead of reduce
```suggestion
    const timestampExcluded = neValues.some((value) => value == accountBalanceTs);
```